### PR TITLE
Add Dark Theme to the QR results Dialog box.

### PR DESCRIPTION
**Pull request by ExypnosPrincess**

### CODE REVIEW

File: app/src/main/java/app/grapheneos/camera/ui/activities/MainActivity.kt [hunk 1]
```diff
@@ -1092,9 +1092,9 @@ open class MainActivity : AppCompatActivity(),
             dialog = Dialog(this, R.style.Theme_Dialog)
             dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
             dialog.setContentView(R.layout.scan_result_dialog)
-            dialog.window?.setBackgroundDrawable(
+            /*dialog.window?.setBackgroundDrawable(
```
Comment (flawedworld) [inline app/src/main/java/app/grapheneos/camera/ui/activities/MainActivity.kt:1095]: Please remove commented code

**Comment by Zanthed** [comment]
Squash the 2 commits into 1 commit

**Comment by MHShetty** [comment]
Hey @ExypnosPrincess,

We might need to make some minor color changes before/after we merge this PR to make it more consistent with the current default theme (for e.g. the result region shading, brighten/whitening of the current result mode) and possibly brighten the copy-to-clipboard and share icon (as they currently look a bit dull in dark theme). I'll merge the PR and make the required changes as soon as I get a more clear idea about this.

Thanks a lot for taking efforts and making this pull request!

Regards,
Mohit

**Comment by ExypnosPrincess** [comment]
Thank you @MHShetty  for your kind reply. I welcome the changes and am aware my initial attempt needed improvements, but wasn't fully sure how to best go forward with them as I am still learning android development at this moment. Thank very much for accepting my changes.

**Comment by ExypnosPrincess** [comment]
Thank you @MHShetty for your kind and informative reply.

File: app/src/main/res/layout/scan_result_dialog.xml [hunk 2]
```diff
@@ -62,13 +65,13 @@
...
         android:maxHeight="200dp"
         android:autoLink="all"
         android:scrollbars="vertical"
         android:fadeScrollbars="false"
         android:overScrollMode="always"
         android:textIsSelectable="true"
-        android:layout_width="match_parent" />
-    
+        android:layout_width="match_parent"
+        android:textColor="@color/qr_results"/>
```
Comment (smdyv) [inline app/src/main/res/layout/scan_result_dialog.xml:75]: This view inherits colors for the text color, so no custom color is needed. If we want a more prominent color than the default, we can use `?android:attr/textColorPrimary`, which is the most prominent text color and a [night-qualified color resource](https://developer.android.com/guide/topics/ui/look-and-feel/darktheme#themes-styles).

```suggestion
        android:textColor="?android:attr/textColorPrimary"/>
```

File: app/src/main/res/layout/scan_result_dialog.xml [hunk 3]
```diff
@@ -62,13 +65,13 @@
         android:layout_marginTop="6dp"
         android:paddingHorizontal="10dp"
         android:paddingVertical="6dp"
-        android:background="@drawable/qr_result_background"
```
Comment (smdyv) [inline app/src/main/res/layout/scan_result_dialog.xml:65]: It has already been expressed by @MHShetty that we want to keep a "box" for the result.

My suggestion:
1. Keep this line.
2. Define a new color element (e.g. `qr_result_background`) with the value of:
    - default: @color/material_dynamic_neutral90
    - night: @color/material_dynamic_neutral20
4. Go to `drawables/qr_result_background.xml` and use the newly defined color resource for the color value (e.g. `@color/qr_result_background`).

**Review by smdyv — CHANGES_REQUESTED** [review]
The result could look like this:

result-light
result-dark

**Comment by ExypnosPrincess** [comment]
Thank so much, @smdyv for taking the time out of your day to explain to me how I could improve my code. I really appreciate it.

**Comment by MHShetty** [comment]
> The result could look like this:
> 
> result-light result-dark

Hey @smdyv,

The new dark theme color combination suggested by you not only looks consistent with the default theme, but also looks really neat and pretty!

Thanks a lot for helping us make this feature better 😊

Regards,
Mohit

**Comment by ExypnosPrincess** [comment]
> > The result could look like this:
> > result-light result-dark
> 
> Hey @smdyv,
> 
> The new dark theme color combination suggested by you not only looks consistent with the default theme, but also looks really neat and pretty!
> 
> Thanks a lot for helping us make this feature better 😊
> 
> Regards, Mohit

I must agree. It does look really pretty. I'm happy they suggested those changes.

**Comment by MHShetty** [comment]
Hey @ExypnosPrincess,

The PR has now been merged to the main branch of our repository. 

Thanks a lot for making this pull request @ExypnosPrincess and @smdyv for those beautiful recommendations!

**Comment by ExypnosPrincess** [comment]
You are very welcome, @MHShetty.

**Pull request merged**